app/services/cache/MongoCacheClient.scala

```diff
@@ -184,6 +189,35 @@ class MongoCacheClient(appConfig: AppConfig, db: () => DefaultDB, applicationCry
       case _ => None
     }
 
+  def findWithCacheMiss[T](id: String, newId: String, key: String)(implicit reads: Reads[T]): Future[Option[T]] = {
+    val newCache: Future[Option[T]] = fetchAll(newId, false) map {
+      case Some(cache) => if (appConfig.mongoEncryptionEnabled) {
+        decryptValue[T](cache, key)(new JsonDecryptor[T](), reads)
+      } else {
+        getValue[T](cache, key)
+      }
+      case _ => None
+    }
+
+    val oldCache: Future[Option[T]] = fetchAll(id, true) map {
+      case Some(cache) => if (appConfig.mongoEncryptionEnabled) {
+        decryptValue[T](cache, key)(new JsonDecryptor[T](), reads)
+      } else {
+        getValue[T](cache, key)
+      }
+      case _ => None
+    }
+
+    for {
```

Comment:
Could use flatten like we did for the countries I think? Easy with respect to this though

Reply:
I would rename old and new to fit in with oid and credId

Reply:
I will refactor this function a bit and we'll see then

Reply:
like above